### DISCUSSION TITLE
Migrating > 1200 in production microservices running across multiple environments (Azure, AWS, GCP, On-premise) from Istio to Linkerd

### DISCUSSION THREAD

**Discussion by nathanawmk** [linkerd/linkerd2#6679] [Q&A]
### What problem are you trying to solve?
Migrating > 1200 in production microservices running across multiple environments (Azure, AWS, GCP, On-premise) from Istio to Linkerd

### How should the problem be solved?
A factory migration approach to migrate from Istio to Linkerd
Allow parallel run of both Istio and Linkerd before shutting down Istio. 

What do you want to happen? Add any considered drawbacks.
Able to monitor my microservices in both Istio and Linkerd before decommissoning Istio

### Any alternatives you've considered?

Is there another way to solve this problem that isn't as good a solution?

### How would users interact with this feature?

If you can, explain how users will be able to use this. Maybe some sample CLI
output?

**Comment by kleimkuhler** [comment]
It's probably worth looking at this [blog post](https://nais.io/blog/posts/2021/05/changing-service-mesh.html)! There is a detailed breakdown about what had to happen for a transition like this to occur without zero downtime.

**Reply by kumandriy** [reply]
in case if somebody stumbles on this, here is a correct [link](https://nais.io/blog/posts/changing-service-mesh/)
